client/app/billing/orders/billing-orders.controller.js

```diff
@@ -1,144 +1,63 @@
-        const search = $location.search();
-        if (search.status) {
-          $scope.activeFilter.status = _.find($scope.statusFilters, {
-            id: search.status,
-          });
-        }
-        if (search.orderId) {
-          $scope.activeFilter.orderId = parseInt(search.orderId, 10);
-        }
+import _ from 'lodash';
+
+export default class BillingOrdersCtrl {
+  /* @ngInject */
+
+  constructor($q, $log, $state, $translate, OvhApiMe,
+    OvhApiMeOrder, constants, orders, schema, criteria, filter) {
+    this.$q = $q;
+    this.$log = $log;
+    this.$state = $state;
+    this.$translate = $translate;
+    this.OvhApiMe = OvhApiMe;
+    this.OvhApiMeOrder = OvhApiMeOrder;
+    this.orders = orders;
+    this.schema = schema;
+    this.criteria = criteria || [];
+    this.filter = filter;
+    this.getSubsidiary().then((subsidiary) => {
+      this.billingGuideUrl = constants.urls[subsidiary].guides.billing;
+    });
+  }
 
-        if (angular.isUndefined($scope.activeFilter.status)) {
-          $scope.activeFilter.status = _.first($scope.statusFilters);
-        }
+  getSubsidiary() {
+    return this.OvhApiMe.v6().get().$promise.then(({ ovhSubsidiary }) => ovhSubsidiary);
+  }
 
-        return BillingOrderStatusEnum.getEnum();
-      })
-      .then((statusEnum) => {
-        $scope.ORDER_STATUS_ENUM = statusEnum;
-      })
-      .then(() => $scope.getOrders());
+  loadRow($row) {
+    return this.OvhApiMeOrder.v6()
+      .getStatus({ orderId: $row.orderId })
+      .$promise
+      .then(status => _.assign($row, status));
   }
 
-  $scope.getOrders = function (forceRefresh) {
-    $scope.orderIds = [];
-    $scope.loaders.orders = true;
-    Alerter.alertFromSWS(null);
+  getStateEnumFilter() {
+    const states = _.get(this.schema.models, 'billing.order.OrderStatusEnum').enum;
+    const filter = {
+      values: {},
+    };
 
-    if ($scope.activeFilter.status) {
-      $location.search('status', $scope.activeFilter.status.id);
-    }
+    states.forEach((state) => {
+      _.set(filter.values, state, this.$translate.instant(`orders_order_status_${state}`));
+    });
 
-    if (forceRefresh) {
-      BillingOrdersApiv7.clearCache();
-    }
+    return filter;
+  }
 
-    let filters = {};
-    if (angular.isFunction($scope.activeFilter.status.getFilter)) {
-      filters = $scope.activeFilter.status.getFilter();
-    }
+  goToOrder({ orderId }) {
+    this.$state.go('app.account.billing.order', {
+      orderId,
+      ordersFilter: this.filter,
+    });
+  }
 
-    if ($scope.activeFilter.orderId) {
-      $location.search('orderId', $scope.activeFilter.orderId);
-      filters['orderId:in'] = $scope.activeFilter.orderId;
+  onCriteriaChange(criteria) {
+    this.criteria = criteria;
+    try {
+      this.filter = encodeURIComponent(JSON.stringify(criteria.map(c => _.omit(c, 'title'))));
+      this.$state.go('app.account.billing.orders', { filter: this.filter }, { reload: false });
```

Comment:
Imo, things related to the `$state` should be extracted in the routing's `resolve` and injected